qt/scientific_interfaces/Muon/MuonAnalysis.cpp

```diff
@@ -449,20 +449,28 @@ std::string MuonAnalysis::addItem(ItemType itemType, int tableRow,
                                   PlotType plotType) {
   AnalysisDataServiceImpl &ads = AnalysisDataService::Instance();
 
-  // Create workspace and a raw (unbinned) version of it
-  auto ws = createAnalysisWorkspace(itemType, tableRow, plotType);
-  auto wsRaw = createAnalysisWorkspace(itemType, tableRow, plotType, true);
-
   // Find names for new workspaces
   const std::string wsName = getNewAnalysisWSName(itemType, tableRow, plotType);
   const std::string wsRawName = wsName + "_Raw";
+  const std::string unnorm = "_unNorm";
+  std::vector<std::string> wsNames = {wsName, wsRawName};
+  // Create workspace and a raw (unbinned) version of it
+  auto ws = createAnalysisWorkspace(itemType, tableRow, plotType, wsName);
+  if (ads.doesExist("tmp_unNorm")) {
+    ads.rename("tmp_unNorm", wsName + unnorm);
+    wsNames.push_back(wsName + unnorm);
+  }
 
+  auto wsRaw =
+      createAnalysisWorkspace(itemType, tableRow, plotType, wsRawName, true);
+  if (ads.doesExist("tmp_unNorm")) {
```

Comment:
Not sure I understand this - do we need to do this check twice? If so, put it in a function.

Reply:
its because tmp_unNorm is produced by createAnalysisWorkspace (the first one is binned and the second is raw/unbinned).

Reply:
If we know it's being produced then why do we need the test, can it fail?

Reply:
yes, it only passes in the case of asymmetry and groups

Reply:
Ok. I'd function it anyway, just so we don't have three lines of code repeated right after each other.